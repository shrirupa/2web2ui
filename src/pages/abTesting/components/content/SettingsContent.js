import React, { Fragment } from 'react';

import { UnstyledLink } from '@sparkpost/matchbox';

const SettingsContent = ({ test }) => (
  <div>
    {
      (test.status === 'draft' || test.status === 'scheduled') && <Fragment>
        <p><span>You may continue to adjust these settings and template variants while this test is in </span>{test.status}<span> mode.</span></p>

        <p><UnstyledLink external to='https://www.sparkpost.com/docs/tech-resources/a-b-testing-sparkpost/'>Learn more about configuring AB tests</UnstyledLink>.</p>
      </Fragment>
    }
  </div>
);

export default SettingsContent;
