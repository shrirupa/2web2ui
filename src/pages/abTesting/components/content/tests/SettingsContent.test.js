import { shallow } from 'enzyme';
import React from 'react';
import SettingsContent from '../SettingsContent';

describe('Settings Content Component', () => {
  let wrapper;
  let props;

  beforeEach(() => {
    props = {
      test: {}
    };
    wrapper = shallow(<SettingsContent {...props} />);
  });

  it('should not render anything by default', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render when test is in draft mode', () => {
    wrapper.setProps({ test: { status: 'draft' }});
    expect(wrapper).toMatchSnapshot();
  });

  it('should render when test is in scheduled mode', () => {
    wrapper.setProps({ test: { status: 'scheduled' }});
    expect(wrapper).toMatchSnapshot();
  });
});
