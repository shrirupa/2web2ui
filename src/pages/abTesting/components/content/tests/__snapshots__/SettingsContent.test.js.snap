// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings Content Component should not render anything by default 1`] = `<div />`;

exports[`Settings Content Component should render when test is in draft mode 1`] = `
<div>
  <React.Fragment>
    <p>
      <span>
        You may continue to adjust these settings and template variants while this test is in 
      </span>
      draft
      <span>
         mode.
      </span>
    </p>
    <p>
      <UnstyledLink
        external={true}
        to="https://www.sparkpost.com/docs/tech-resources/a-b-testing-sparkpost/"
      >
        Learn more about configuring AB tests
      </UnstyledLink>
      .
    </p>
  </React.Fragment>
</div>
`;

exports[`Settings Content Component should render when test is in scheduled mode 1`] = `
<div>
  <React.Fragment>
    <p>
      <span>
        You may continue to adjust these settings and template variants while this test is in 
      </span>
      scheduled
      <span>
         mode.
      </span>
    </p>
    <p>
      <UnstyledLink
        external={true}
        to="https://www.sparkpost.com/docs/tech-resources/a-b-testing-sparkpost/"
      >
        Learn more about configuring AB tests
      </UnstyledLink>
      .
    </p>
  </React.Fragment>
</div>
`;
